packages/@lwc/synthetic-shadow/src/shared/static-html-collection.ts

```diff
@@ -4,17 +4,13 @@
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
-import { ArrayMap, create, defineProperty, fields, forEach, setPrototypeOf } from '@lwc/shared';
+import { create, defineProperty, fields, forEach, setPrototypeOf } from '@lwc/shared';
 const { createFieldName, getHiddenField, setHiddenField } = fields;
 
-const Items = createFieldName('items');
+const Items = createFieldName('StaticHTMLCollection', 'items');
 
-function isValidHTMLCollectionName(name) {
-    return name !== 'length' && isNaN(name);
-}
-
-function getNodeHTMLCollectionName(node) {
-    return node.getAttribute('id') || node.getAttribute('name');
+function getNodeHTMLCollectionName(element: Element) {
```

Comment:
the name of the method is still confusing, maybe `getHTMLCollectionElementName()`